p2p/src/config.rs

```diff
@@ -15,6 +15,50 @@
 
 use serde::{Deserialize, Serialize};
 
+/// Multicast DNS configuration.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(tag = "type")]
+pub enum MdnsConfig {
+    Enabled {
+        /// Interval (in milliseconds) at which to poll the network for new peers.
+        query_interval: u64,
+        /// Use IPv6 for multicast DNS
+        enable_ipv6_mdns_discovery: bool,
+    },
+    Disabled,
+}
+
+impl MdnsConfig {
+    pub fn new() -> Self {
+        MdnsConfig::Disabled
+    }
+
+    pub fn from_options(
+        enable_mdns: bool,
+        query_interval: Option<u64>,
+        enable_ipv6_mdns_discovery: Option<bool>,
+    ) -> Self {
+        if enable_mdns {
+            MdnsConfig::Enabled {
+                query_interval: query_interval.unwrap_or(0),
```

Comment:
Can you please put the default values of the configuration somewhere isolated, instead of directly injecting them to options?